### PR TITLE
Student tests

### PR DESCRIPTION
**Pull request by Jokeswar**

### CODE REVIEW

File: testsuite/test_student_workflow.py [hunk 1]
```diff
@@ -66,10 +74,142 @@ def test_submission(client, live_server, base_db_setup):
...
+    )
+
+    response = client.get(f'/submission/{submission.pk}/download')
+    assert response.status_code == 403
+
+
+@pytest.mark.django_db
+def test_user_cannot_review(client, STC, base_db_setup):
+    (user, course, assignment) = base_db_setup
+    user.is_staff = False
```
Comment (gmuraru) [inline testsuite/test_student_workflow.py:131]: You can remove this and the next line

File: testsuite/test_student_workflow.py [hunk 2]
```diff
@@ -66,10 +74,142 @@ def test_submission(client, live_server, base_db_setup):
...
+    STC.assertRedirects(
+        response,
+        f'/admin/login/?next=/submission/{submission.pk}/review',
+    )
+
+
+@pytest.mark.django_db
+def test_user_cannot_recompute(client, STC, base_db_setup):
+    (user, course, assignment) = base_db_setup
+    user.is_staff = False
```
Comment (gmuraru) [inline testsuite/test_student_workflow.py:155]: You can remove this and the next line

File: testsuite/test_student_workflow.py [hunk 3]
```diff
@@ -66,10 +74,142 @@ def test_submission(client, live_server, base_db_setup):
...
+    STC.assertRedirects(
+        response,
+        f'/admin/login/?next=/submission/{submission.pk}/recompute',
+    )
+
+
+@pytest.mark.django_db
+def test_user_cannot_rerun(client, STC, base_db_setup):
+    (user, course, assignment) = base_db_setup
+    user.is_staff = False
```
Comment (gmuraru) [inline testsuite/test_student_workflow.py:175]: You can remove this and the next line

File: testsuite/test_submission_actions.py [hunk 4]
```diff
@@ -45,16 +44,17 @@ def test_review(client, base_db_setup):
 
 
 def test_recompute(client, base_db_setup):
-    (user, course, assignment) = base_db_setup
-    client.login(username='user', password='pw')
-    course.teaching_assistants.add(user)
+    (_, user, _, course, assignment) = base_db_setup
```
Comment (gmuraru) [inline testsuite/test_submission_actions.py:47]: Here ```user``` is the ```ta```

Comment (Jokeswar) [reply testsuite/test_submission_actions.py:47]: Addressed by #222

**Pull request merged**